## Checeking the goodness of fit for a normal distribution::

In [1]:
import numpy as np
import scipy.stats

In [2]:
## example data
A = [33,43,44,45,52,56,58,63,64,65,66,68,70,72,73,74,75,83,84,85,86,91,92,94,98,102,105]

In [3]:
mean = np.mean(A)
mean

71.88888888888889

In [4]:
std = np.std(A)
std

18.787965678087335

In [5]:
x = 1/6   ## for 6 equal probability intervals

In [6]:
for j in range(1,6):
    prob_intervals = [scipy.stats.norm.ppf(j*x,mean,std)]
    print(prob_intervals)

[53.71300570872863]
[63.79639917311058]
[71.88888888888889]
[79.98137860466719]
[90.06477206904914]


In [7]:
exp_f = [5,5,5,5,5,5]  ## will divide the normal distribution into 6 intervals at frequency 5 in each

In [8]:
obs_f = [6,3,6,5,4,6]
## In interval 0 to 53.71 we have 6 data
## In interval 53.71 to 63.79 we have 3 data and so on.....

In [9]:
scipy.stats.chisquare(obs_f,exp_f)

Power_divergenceResult(statistic=1.5999999999999999, pvalue=0.9012493445012737)

## Since, value=0.90 > alpha =0.05
# This Implies ==> accept Null hypothesis 
    i.e. our data will follow a normal distribution